app/services/s3_query_service.rb

```diff
@@ -325,16 +325,7 @@ def md5(io:)
 
   def count_objects(bucket_name: self.bucket_name, prefix: self.prefix)
     responses = s3_responses(bucket_name:, prefix:)
-    total_key_count = responses.reduce(0) { |total, resp| total + resp.key_count }
-    if total_key_count == 0
-      # if the bucket does not exist
-      # it will return 0
-      0
-    else
-      # if the bucket does exist
-      # s3 always sends back the bucket key as the first response, so we should not count it
-      total_key_count - 1
-    end
+    responses.reduce(0) { |total, resp| total + resp.key_count }
```

Comment:
AWS seems to return the correct key_counts and we don't need to subtract 1 from the total anymore. We tested with a bucket that does not exist and with a bucket that exists (and has files). It is not possible to have an empty bucket so that we could not test.